Remove the unused Surface param from convert alpha docs

MyreMylar opened the pull request:
It does nothing at all.

fixes #1753

zoldalma999 commented:
Apart from changing the docs, we could also remove the stubs for it and deprecate it in code. That would also close #493. Unless we want to add this functionality back?

MyreMylar commented:
> Apart from changing the docs, we could also remove the stubs for it and deprecate it in code. That would also close #493. Unless we want to add this functionality back?

Pretty sure we don't want this back.

I've added the deprecation warnings.

Temmie3754 reviewed (approved):
No longer appears in docs, stubs and appropriate depreciation warning displays

ankith26 commented on docs/reST/ref/surface.rst:
```diff
@@ -240,17 +240,15 @@
+      be optimized for fast alpha blitting to it.
 
       As with :meth:`convert()` the returned surface has the same class as
       the converted surface.
```
A note here of the deprecation would be nice

MyreMylar replied:
👍 added a `versionchanged` note.

ankith26 reviewed (changes requested):
Also a small and simple test that tests the deprecation warning would also be nice

MyreMylar commented:
> Also a small and simple test that tests the deprecation warning would also be nice

Added a test for the deprecation warning, and removed a few places in the test that were now triggering it as they were passing in the surface argument (which has done nothing for years).

zoldalma999 commented on test/surface_test.py:
```diff
@@ -11,6 +11,7 @@
 except (ImportError, NameError):
     pass
 import pygame
+import pygame
```
this seems accidental

MyreMylar replied:
oops, thanks 👍 

Fixed.

zoldalma999 reviewed (approved):
Apart from a minor nitpick, this looks good to me

MyreMylar commented on test/surface_test.py:
```diff
@@ -11,6 +11,7 @@
 except (ImportError, NameError):
     pass
 import pygame
+import pygame
```
```suggestion
```

ankith26 reviewed (approved):
LGTM now, thanks for the PR :tada:
